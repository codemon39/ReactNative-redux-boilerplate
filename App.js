import React, {Component} from 'react';
import {Provider} from 'react-redux';

import {store} from './app/store/store';
import {ThemeProvider} from './app/theme/useTheme';
import {NoInternetToast} from './app/components/NoInternet';

// Navigation
import RootNavigation from './app/routes/RootNavigation';

export default class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <ThemeProvider>
          <RootNavigation />
          <NoInternetToast />
        </ThemeProvider>
      </Provider>
    );
  }
}
