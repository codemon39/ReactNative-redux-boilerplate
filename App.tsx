import React from 'react';
import {SafeAreaProvider} from 'react-native-safe-area-context';

import ReduxProvider from './app/store';
import {STORYBOOK_MODE} from '@env';
import {ThemeProvider} from './app/theme/useTheme';
import {NoInternetToast} from './app/components/NoInternet';

// Navigation
import RootNavigation from './app/routes/RootNavigation';

import StorybookUIRoot from './.storybook';

let Root = function App() {
  return (
    <SafeAreaProvider>
      <ReduxProvider>
        <ThemeProvider>
          <RootNavigation />
          <NoInternetToast />
        </ThemeProvider>
      </ReduxProvider>
    </SafeAreaProvider>
  );
};

// Render StoryBook if the ENV variable set to 'TRUE', type is <string> not <boolean>
if (STORYBOOK_MODE === 'TRUE') {
  Root = StorybookUIRoot;
}

export default Root;
