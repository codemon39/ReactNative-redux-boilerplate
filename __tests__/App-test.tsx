/**
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */
import 'react-native';
import React from 'react';
import {render} from '@testing-library/react-native';
import App from '../App';
jest.useFakeTimers();

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('App', () => {
  it('renders correctly', () => {
    renderer.create(<App />);
  });

  it('should renders correctly', () => {
    const page = render(<App />);
    expect(page.toJSON()).toMatchSnapshot();
  });
});
