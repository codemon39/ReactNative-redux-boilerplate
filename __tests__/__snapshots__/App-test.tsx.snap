// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should renders correctly 1`] = `
Array [
  <RNCSafeAreaView
    edges={
      Array [
        "bottom",
        "left",
        "right",
        "top",
      ]
    }
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNCSafeAreaProvider
      onInsetsChange={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenNavigationContainer
        style={
          Object {
            "flex": 1,
            "overflow": "hidden",
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            style={
              Array [
                Object {
                  "backgroundColor": "rgb(242, 242, 242)",
                  "flex": 1,
                },
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column-reverse",
                  },
                  undefined,
                ],
              ]
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#ffffff",
                      "flex": 1,
                    },
                    Object {
                      "backgroundColor": "#f0f0f0",
                    },
                    undefined,
                  ]
                }
              >
                <RCTScrollView
                  contentContainerStyle={
                    Object {
                      "paddingHorizontal": 12,
                      "paddingVertical": 16,
                    }
                  }
                  data={
                    Array [
                      Object {
                        "done": false,
                        "id": "1",
                        "title": "Task 1",
                      },
                      Object {
                        "done": false,
                        "id": "2",
                        "title": "Task 2",
                      },
                    ]
                  }
                  getItem={[Function]}
                  getItemCount={[Function]}
                  keyExtractor={[Function]}
                  onContentSizeChange={[Function]}
                  onLayout={[Function]}
                  onMomentumScrollBegin={[Function]}
                  onMomentumScrollEnd={[Function]}
                  onScroll={[Function]}
                  onScrollBeginDrag={[Function]}
                  onScrollEndDrag={[Function]}
                  removeClippedSubviews={false}
                  renderItem={[Function]}
                  scrollEventThrottle={50}
                  stickyHeaderIndices={Array []}
                  viewabilityConfigCallbackPairs={Array []}
                >
                  <View>
                    <View
                      onLayout={[Function]}
                      style={null}
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#ffffff",
                              "borderRadius": 8,
                              "paddingHorizontal": 16,
                              "paddingVertical": 16,
                              "width": "100%",
                            },
                            Object {
                              "backgroundColor": "#ffffff",
                            },
                            Object {
                              "marginBottom": 10,
                            },
                          ]
                        }
                      >
                        <View
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "justifyContent": "space-between",
                              },
                              Object {
                                "opacity": 1,
                              },
                            ]
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "#4d505b",
                                  "fontSize": 14,
                                  "fontWeight": "600",
                                },
                                Object {
                                  "color": "rgb(58, 58, 60)",
                                  "textDecorationLine": "none",
                                },
                              ]
                            }
                          >
                            Task 1
                          </Text>
                          <Text
                            allowFontScaling={false}
                            selectable={false}
                            style={
                              Array [
                                Object {
                                  "color": "gray",
                                  "fontSize": 20,
                                },
                                undefined,
                                Object {
                                  "fontFamily": "Ionicons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            
                          </Text>
                        </View>
                      </View>
                    </View>
                    <View
                      onLayout={[Function]}
                      style={null}
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#ffffff",
                              "borderRadius": 8,
                              "paddingHorizontal": 16,
                              "paddingVertical": 16,
                              "width": "100%",
                            },
                            Object {
                              "backgroundColor": "#ffffff",
                            },
                            Object {
                              "marginBottom": 10,
                            },
                          ]
                        }
                      >
                        <View
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "justifyContent": "space-between",
                              },
                              Object {
                                "opacity": 1,
                              },
                            ]
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "#4d505b",
                                  "fontSize": 14,
                                  "fontWeight": "600",
                                },
                                Object {
                                  "color": "rgb(58, 58, 60)",
                                  "textDecorationLine": "none",
                                },
                              ]
                            }
                          >
                            Task 2
                          </Text>
                          <Text
                            allowFontScaling={false}
                            selectable={false}
                            style={
                              Array [
                                Object {
                                  "color": "gray",
                                  "fontSize": 20,
                                },
                                undefined,
                                Object {
                                  "fontFamily": "Ionicons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </RCTScrollView>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#ffffff",
                        "borderRadius": 8,
                        "paddingHorizontal": 16,
                        "paddingVertical": 16,
                        "width": "100%",
                      },
                      Object {
                        "backgroundColor": "#ffffff",
                      },
                      Array [
                        Object {
                          "borderBottomLeftRadius": 0,
                          "borderBottomRightRadius": 0,
                          "borderTopWidth": 1,
                          "elevation": 4,
                          "shadowColor": "#000",
                          "shadowOffset": Object {
                            "height": 2,
                            "width": 0,
                          },
                          "shadowOpacity": 0.23,
                          "shadowRadius": 2.62,
                        },
                        Object {
                          "borderTopColor": "#f0f0f0",
                        },
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                      }
                    }
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "borderColor": "#c50e29",
                            "borderRadius": 20,
                            "borderWidth": 0.5,
                            "justifyContent": "center",
                            "marginRight": 8,
                            "paddingHorizontal": 5,
                            "paddingVertical": 5,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                        ]
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#c50e29",
                              "fontSize": 16,
                            },
                            undefined,
                            Object {
                              "fontFamily": "Ionicons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                      >
                        
                      </Text>
                    </View>
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }
                    >
                      <TextInput
                        onChangeText={[Function]}
                        onSubmitEditing={[Function]}
                        placeholder="New Task"
                        placeholderTextColor="rgb(58, 58, 60)"
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#f6f6f6",
                              "borderColor": "#f0f0f0",
                              "borderRadius": 20,
                              "borderWidth": 0.5,
                              "flex": 1,
                              "fontSize": 14,
                              "height": 38,
                              "paddingHorizontal": 10,
                            },
                            Object {
                              "backgroundColor": "#f0f0f0",
                              "color": "rgb(58, 58, 60)",
                            },
                          ]
                        }
                        value=""
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "zIndex": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "backgroundColor": "#ffffff",
                      "borderBottomColor": "rgb(216, 216, 216)",
                      "flex": 1,
                      "shadowColor": "rgb(216, 216, 216)",
                      "shadowOffset": Object {
                        "height": 0.5,
                        "width": 0,
                      },
                      "shadowOpacity": 0.85,
                      "shadowRadius": 0,
                    }
                  }
                />
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "height": 50,
                    "maxHeight": undefined,
                    "minHeight": undefined,
                    "opacity": undefined,
                    "transform": undefined,
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "height": 0,
                    }
                  }
                />
                <View
                  pointerEvents="box-none"
                  style={
                    Object {
                      "alignItems": "stretch",
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    collapsable={false}
                    pointerEvents="box-none"
                    style={
                      Object {
                        "alignItems": "flex-start",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "justifyContent": "center",
                        "marginStart": 0,
                      }
                    }
                  />
                  <View
                    collapsable={false}
                    pointerEvents="box-none"
                    style={
                      Object {
                        "justifyContent": "center",
                        "marginHorizontal": 16,
                        "maxWidth": 718,
                      }
                    }
                  >
                    <Text
                      accessibilityRole="header"
                      aria-level="1"
                      collapsable={false}
                      numberOfLines={1}
                      style={
                        Object {
                          "color": "rgb(58, 58, 60)",
                          "fontSize": 16,
                          "fontWeight": "600",
                        }
                      }
                    >
                      To Do
                    </Text>
                  </View>
                  <View
                    collapsable={false}
                    pointerEvents="box-none"
                    style={
                      Object {
                        "alignItems": "flex-end",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "justifyContent": "center",
                        "marginEnd": 0,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenNavigationContainer>
      <View
        collapsable={false}
        onLayout={[Function]}
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderTopColor": "rgb(216, 216, 216)",
            "borderTopWidth": 0.5,
            "bottom": 0,
            "elevation": 8,
            "height": 49,
            "left": 0,
            "paddingBottom": 0,
            "paddingHorizontal": 0,
            "position": null,
            "right": 0,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View
          accessibilityRole="tablist"
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityLabel="To Do, tab, 1 of 2"
            accessibilityRole="button"
            accessibilityState={
              Object {
                "selected": true,
              }
            }
            accessibilityStates={
              Array [
                "selected",
              ]
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "flexDirection": "column",
                  "justifyContent": "flex-end",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "minWidth": 25,
                      "position": "absolute",
                      "width": "100%",
                    },
                    Object {
                      "opacity": 1,
                    },
                  ]
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#0071ff",
                        "fontSize": 20,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "minWidth": 25,
                      "position": "absolute",
                      "width": "100%",
                    },
                    Object {
                      "opacity": 0,
                    },
                  ]
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "rgb(58, 58, 60)",
                        "fontSize": 20,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#0071ff",
                  },
                  Object {
                    "fontSize": 10,
                  },
                  undefined,
                ]
              }
            >
              To Do
            </Text>
          </View>
          <View
            accessibilityLabel="Settings, tab, 2 of 2"
            accessibilityRole="button"
            accessibilityState={
              Object {
                "selected": false,
              }
            }
            accessibilityStates={Array []}
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "flexDirection": "column",
                  "justifyContent": "flex-end",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "minWidth": 25,
                      "position": "absolute",
                      "width": "100%",
                    },
                    Object {
                      "opacity": 0,
                    },
                  ]
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#0071ff",
                        "fontSize": 20,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "height": "100%",
                      "justifyContent": "center",
                      "minWidth": 25,
                      "position": "absolute",
                      "width": "100%",
                    },
                    Object {
                      "opacity": 1,
                    },
                  ]
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "rgb(58, 58, 60)",
                        "fontSize": 20,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "textAlign": "center",
                  },
                  Object {
                    "color": "rgb(58, 58, 60)",
                  },
                  Object {
                    "fontSize": 10,
                  },
                  undefined,
                ]
              }
            >
              Settings
            </Text>
          </View>
        </View>
      </View>
    </RNCSafeAreaProvider>
  </RNCSafeAreaView>,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#d70015",
        "bottom": 0,
        "flexDirection": "row",
        "height": 18,
        "justifyContent": "center",
        "position": "absolute",
        "width": 750,
        "zIndex": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 11,
        }
      }
    >
      No Internet Connection
    </Text>
  </View>,
]
`;
